modules/gdpr_fields/src/Entity/GdprField.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Drupal\gdpr_fields\Entity;
+
+/**
+ * Metadata for a GDPR field.
+ */
+class GdprField {
+
+  /**
+   * Bundle name.
+   *
+   * @var string
+   */
+  public $bundle;
+
+  /**
+   * Field name.
+   *
+   * @var string
+   */
+  public $name;
+
+  /**
+   * Wether GDPR is enabled for this field.
+   *
+   * @var bool
+   */
+  public $enabled = FALSE;
+
+  /**
+   * Right to Forget setting for this field.
+   *
+   * @var string
+   */
+  public $rtf = 'no';
+
+  /**
+   * Right to Access setting for this field.
+   *
+   * @var string
+   */
+  public $rta = 'no';
+
+  /**
+   * Sanitizer to use on this field.
+   *
+   * @var string
+   */
+  public $sanitizer = '';
+
+  /**
+   * Notes.
+   *
+   * @var string
+   */
+  public $notes = '';
+
+  /**
+   * Whether this field has been configured for GDPR.
+   *
+   * This is different for enabled -
+   * something can be configured but not enabled.
+   *
+   * @var bool
+   */
+  public $configured = FALSE;
+
+  /**
+   * GdprField constructor.
+   *
+   * @param string $bundle
+   *   Bundle.
+   * @param string $name
+   *   Field name.
+   */
+  public function __construct(string $bundle, string $name) {
+    $this->bundle = $bundle;
+    $this->name = $name;
+  }
+
+  /**
+   * Creates a GdprField instance based on array data from the config entity.
+   *
+   * @param array $array
+   *   The undlerying data.
+   *
+   * @return \Drupal\gdpr_fields\Entity\GdprField
+   *   The field metadata instance.
+   */
+  public static function create(array $array) {
+    $field = new GdprField($array['bundle'], $array['name']);
+    $field->rtf = $array['rtf'];
```

Comment:
Lets call `$array` `$values` instead, and use `new static($values['bundle'] ... )` as we don't need to explicitly name the class in a static method.

Reply:
Done

Reply:
Did you push the branch? 😛

Reply:
Yes, I pushed develop. I'll merge it to here too.